code/modules/projectiles/guns/ballistic/rifle.dm

```diff
@@ -71,6 +74,39 @@
 		can_bayonet = FALSE
 		update_icon()
 
+/obj/item/gun/ballistic/rifle/boltaction/attack_self(mob/user)
+	if(jammed)
+		if(prob(unjam_chance))
+			jammed = FALSE
+			unjam_chance = 10
+		else
+			unjam_chance += 10
+			to_chat(user, "<span class='warning'>[src] is jammed!</span>")
+			return FALSE
+	..()
+
+/obj/item/gun/ballistic/rifle/boltaction/process_fire(mob/user)
+	if(chambered.BB)
+		jamming_chance  += 5
+	if(prob(jamming_chance))
+		jammed = TRUE
+	..()
+
+/obj/item/gun/ballistic/rifle/boltaction/Initialize()
+	. = ..()
+	RegisterSignal(src, COMSIG_PARENT_ATTACKBY, .proc/try_clean_weapon)
+
+/obj/item/gun/ballistic/rifle/boltaction/proc/check_attack_item(datum/source, obj/item/item, mob/user, params)
+	SIGNAL_HANDLER
+	if(istype(item, /obj/item/gun_maintenance_supplies))
+		INVOKE_ASYNC(src, .proc/try_clean_weapon, item, user)
+
+/obj/item/gun/ballistic/rifle/boltaction/proc/try_clean_weapon(datum/source, obj/item/item, mob/user, params)
+	if(do_after(user, 10 SECONDS, target = source))
+		user.visible_message("<span class='notice'>[user] finishes maintenance of [source].</span>")
+		jamming_chance = 10
```

Comment:
is this intended to be lower than the default jamming chance?

Reply:
yeah, when mosins come in, the description says they feel moist, so I thought you could do some maintenance on them first, they were unmaintained for a long time after all ;)